src/main/java/com/aws/greengrass/mqttbridge/clients/IoTCoreClient.java

```diff
@@ -129,12 +133,33 @@ public synchronized void updateSubscriptions(Set<String> topics, @NonNull Consum
         topicsToSubscribe.removeAll(subscribedIotCoreTopics);
 
         topicsToSubscribe.forEach(s -> {
-            try {
-                subscribeToIotCore(s);
-                LOGGER.atDebug().kv(TOPIC, s).log("Subscribed to topic");
-                subscribedIotCoreTopics.add(s);
-            } catch (InterruptedException | ExecutionException | TimeoutException e) {
-                LOGGER.atError().kv(TOPIC, s).log("Failed to subscribe");
+            while (!Thread.currentThread().isInterrupted()) {
+                try {
+                    subscribeToIotCore(s);
+                    LOGGER.atDebug().kv(TOPIC, s).log("Subscribed to topic");
```

Comment:
remove, we log this already

Reply:
removed this and similar log from other MQTT client.